File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.js
function solution(numbers, direction) {
    if (direction == 'right') {
        // 1
        // numbers.unshift(numbers.pop())
        
        // 2
        numbers = [numbers.pop(), ...numbers]
    } else {
        // 1
        // numbers.push(numbers.shift())
        
        // 2
        numbers = [...numbers.slice(1), numbers.shift()]
    }
    return numbers;
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.py
from collections import deque

def solution(numbers, direction):
    # 1
    # if direction == 'right' :
    #     return [numbers[-1]] + numbers[:-1]
    # else :
    #     return numbers[1:] + [numbers[0]]
    
    # 2
    if direction == 'right' :
        numbers = deque(numbers)
        numbers.rotate(1) 
    else :
        numbers = deque(numbers)
        numbers.rotate(-1) 
    return list(numbers)